docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,538 @@
+
+        - **`condition`** *string*
+
+          This REQUIRED property specify the compatibility validation rule.
+          Allowed values: _all_ - all referenced graphs must pass; _one_ - one referenced graph must pass from the list.
+
+        - **`annotations`** *string-string map*
+
+          This OPTIONAL property describes user defined annotations.
+
+Example of use:
+
+- Simple requirements - container supporting only NVIDIA GPUs
+
+```json
+{
+  "spec": {
+    "compatibilities": [
+      {
+        "id": "nvidiaGPU",
+        "domain": "org.opencontainers",
+        "attributes": {
+          "hardware.pci.class-id": "0380",
+          "hardware.pci.vendor-id": "10de",
+          "kernel.configuration": [
+            "CONFIG_HOTPLUG_PCI_PCIE=y",
+            "CONFIG_MODULES=y",
+            "CONFIG_MODULE_UNLOAD=y",
+            "CONFIG_PCI_MMCONFIG=m",
+            "CONFIG_DRM_NOUVEAU=n"
+          ]
+        }
+      }
+    ]
+  }
+}
+```
+
+- Complex requirements with relations - container supporting multiple GPUs
+
+```json
+{
+  "spec": {
+    "compatibilities": [
+      {
+        "id": "intelCpu",
+        "domain": "org.opencontainers",
+        "attributes": {
+          "hardware.cpu.vendor": "GenuineIntel",
+          "hardware.cpu.virtualization": "VT-x",
+          "kernel.cmdline": ["intel_iommu=on"]
+        }
+      },
+      {
+        "id": "amdCpu",
+        "domain": "org.opencontainers",
+        "attributes": {
+          "hardware.cpu.vendor": "AuthenticAMD",
+          "hardware.cpu.virtualization": "AMD-V",
+          "kernel.cmdline": ["amd_iommu=pt"]
+        }
+      },
+      {
+        "id": "vfio",
+        "domain": "org.opencontainers",
+        "attributes": {
+          "kernel.modules": [
+            {"name": "vfio"},
+            {"name": "vfio_iommu_type1"},
+            {"name": "vfio-pci"},
+          ]
+        }
+      },
+      {
+        "id": "intelGpu",
+        "domain": "org.opencontainers",
+        "attributes": {
+          "hardware.pci.class-id": "0380",
+          "hardware.pci.vendor-id": "8086",
+          "kernel.configuration": [
+            "CONFIG_DRM=y",
+            "CONFIG_DRM_I915=y",
+            "CONFIG_DRM_I915_USERPTR=y",
+            "CONFIG_DRM_I915_GVT=m",
+            "CONFIG_DRM_I915_GVT_KVMGT=m"
+          ]
+        }
+      },
+       {
+        "id": "nvidiaGPU",
+        "domain": "org.opencontainers",
+        "attributes": {
+          "hardware.pci.class-id": "0380",
+          "hardware.pci.vendor-id": "10de",
+          "kernel.configuration": [
+            "CONFIG_HOTPLUG_PCI_PCIE=y",
+            "CONFIG_MODULES=y",
+            "CONFIG_MODULE_UNLOAD=y",
+            "CONFIG_PCI_MMCONFIG=m",
+            "CONFIG_DRM_NOUVEAU=n"
+          ]
+        }
+      }
+    ],
+    "relations": {
+      "graphs": {
+        "intel": {
+          "edges": [
```

Comment:
If there is going to be a graph in here I think we might consider using an already defined schema, and not make up our own.

Reply:
I am against incorporating external schema to the official spec.
With in-house spec we get the full control (which is the most important IMO), flexibility and customization.

Additionally, the context is a little bit different here, it's not only a graph representation, but the graph with validation context expressed over `all`, `oneOf`, `noneOf` conditions.